### DISCUSSION TITLE
<b-rate> does not update binding value

### DISCUSSION THREAD

**Discussion by rnanda-gh** [buefy/buefy#3730] [Q&A]
I have the following code using <b-rate> but the binding value does not update on '@change' event even though the 'show-score' shows correct rating.

<b-rate
    v-model='rating'
    icon-pack='fa'
    icon='fa-solid fa-star'
    show-score
    @change='showRating'>
</b-rate>

....

        data() {
            return {
                rating: 3.0,  // Default rating
            }
        },

showRating() {
    let msg = "Thank for your rating the post " + this.rating + " stars!"
    this.$buefy.toast.open({
        message: msg,
        type: 'is-success'
    })
}

**Comment by GMartigny** [comment]
[I can't reproduce](https://codepen.io/GMartigny/pen/rNJYwvZ?editors=1010).

**Reply by rnanda-gh** [reply]
It is clearly reproducible. If you click on a different rating (say 5), the alert will show "Thank you for rating the post 3 stars!"

The alert always shows the previous rating and not the current rating.

**Reply by GMartigny** [reply]
Hummm I see what you mean. The `"change"` event is [triggered before](https://github.com/buefy/buefy/blob/dev/src/components/rate/Rate.vue#L122-L123) the `"input"` event that change the value.

Switching both event could be considered breaking change, so I'm not sure it could be easily done.

A quickfix would be to use the value sent bye the event:

```js
        showRating(newValue) {
            let msg = "Thank for your rating the post " + newValue + " stars!";
            this.$buefy.toast.open({
                message: msg,
                type: "is-success"
            });
        }
```
